stacks/api/bucket/history/history.acceptance.spec.ts

```diff
@@ -0,0 +1,210 @@
+import {TestingModule, Test} from "@nestjs/testing";
+import {PassportTestingModule} from "@spica-server/passport/testing";
+import {DatabaseTestingModule, DatabaseService, ObjectId} from "@spica-server/database/testing";
+import {HistoryModule} from "./history.module";
+
+describe("History Acceptance", () => {
+  let app: INestApplication;
+  let req: Request;
+  let module: TestingModule;
+
+  const bucketId = new ObjectId();
+  const documentId = new ObjectId();
+  const anotherDocumentId = new ObjectId();
+
+  const firstHistoryId = new ObjectId(
+    Math.floor(new Date(2018, 5, 10).getTime() / 1000).toString(16) + "0000000000000000"
+  );
+  const secondHistoryId = new ObjectId(
+    Math.floor(new Date(2018, 5, 11).getTime() / 1000).toString(16) + "0000000000000000"
+  );
+
+  beforeAll(async () => {
+    module = await Test.createTestingModule({
+      imports: [
+        CoreTestingModule,
+        PassportTestingModule.initialize(),
+        DatabaseTestingModule.replicaSet(),
+        HistoryModule
+      ]
+    }).compile();
+    app = module.createNestApplication();
+    req = module.get(Request);
+    await app.listen(req.socket);
+
+    //add bucket
+    await app
+      .get(DatabaseService)
+      .collection("buckets")
+      .insertOne({
+        _id: bucketId,
+        title: "New Bucket",
+        description: "Describe your new bucket",
+        icon: "view_stream",
+        primary: "title",
+        readOnly: false,
+        properties: {
+          title: {
+            type: "string",
+            title: "title",
+            description: "Title of the row",
+            options: {position: "left", visible: true}
+          },
+          description: {
+            type: "string",
+            title: "description",
+            description: "Description of the row",
+            options: {position: "right"}
+          }
+        }
+      });
+    //add documents
+    await app
+      .get(DatabaseService)
+      .collection(`bucket_${bucketId}`)
+      .insertMany([
+        //we need to keep last updated data
+        {
+          _id: documentId,
+          title: "last updated title",
+          description: "last updated description"
+        },
+        {
+          _id: anotherDocumentId,
+          title: "another title",
+          description: "another description"
+        }
+      ]);
+    //add histories
+
+    await app
+      .get(DatabaseService)
+      .collection("history")
+      .insertMany([
+        {
+          _id: firstHistoryId,
+          bucket_id: bucketId,
+          document_id: documentId,
+          changes: [
+            {
+              kind: 1,
+              path: ["title"],
+              patches: [
+                {
+                  diffs: [[-1, "updated"], [1, "current"], [0, " tit"]],
+                  start1: 0,
+                  start2: 0,
+                  length1: 11,
+                  length2: 11
+                }
+              ]
+            },
+            {
```

Comment:
why do you insert the changes manually?

Reply:
Because you said 'Do not use diff methods, use more spesific data' :)